kukur/source/piwebapi_af/piwebapi_af.py

```diff
@@ -100,6 +108,14 @@ def __init__(self, config: Dict):
         if "username" in config and "password" in config:
             self.__basic_auth = (config["username"], config["password"])
 
+        self.__oidc_config = None
+        if (
+            (client_id := config.get("client_id"))
+            and (client_secret := config.get("client_secret"))
+            and (oidc_token_url := config.get("oidc_token_url"))
+        ):
+            self.__oidc_config = OIDCConfig(client_id, client_secret, oidc_token_url)
```

Comment:
Instead of repeating this 3 times, can we have an `OIDCConfig.from_config(data: Dict) -> Optional[OIDCConfig]` method?